samples/algorithms/sudoku-grover/ColoringGroverWithConstraints.qs

```diff
@@ -319,51 +149,30 @@ namespace Microsoft.Quantum.Samples.ColoringGroverWithConstraints {
     /// The number of Vertices in the graph.
     /// ## bitsPerColor
     /// The number of bits per color.
-    /// ## maxIterations
+    /// ## nIterations
     /// An estimate of the maximum iterations needed.
     /// ## oracle
     /// The Oracle used to find solution.
+    /// ## statePrep
+    /// Routine that prepares an equal superposition of all basis states in the search space.
     ///
     /// # Output
     /// Int Array giving the color of each vertex.
     ///
-    /// # Remarks
-    /// See https://github.com/microsoft/QuantumKatas/tree/main/SolveSATWithGrover
-    /// for original implementation in SolveSATWithGrover Kata.
-    operation FindColorsWithGrover (numVertices : Int, bitsPerColor : Int, maxIterations : Int,
-        oracle : ((Qubit[], Qubit) => Unit is Adj)) : Int[] {
-        // This task is similar to task 2.2 from SolveSATWithGrover kata,
-        // but the percentage of correct solutions is potentially higher.
-        mutable coloring = new Int[numVertices];
+    operation FindColorsWithGrover(
+        numVertices : Int, bitsPerColor : Int, nIterations : Int,
+        oracle : ((Qubit[], Qubit) => Unit is Adj),
+        statePrep : (Qubit[] => Unit is Adj)) : Int[] {
 
         // Note that coloring register has the number of qubits that is
         // twice the number of vertices (bitsPerColor qubits per vertex).
         use register = Qubit[bitsPerColor * numVertices];
-        use output = Qubit();
 
-        mutable correct = false;
-        mutable iter = 1;
-        // Try for one iteration, if it fails, try again for one more iteration and repeat until maxIterations is reached.
-        repeat {
-            Message($"Trying search with {iter} iterations...");
-            ApplyGroversAlgorithmLoop(register, oracle, iter);
-            let res = MultiM(register);
-            // to check whether the result is correct, apply the oracle to the
-            // register plus auxiliary after measurement.
-            oracle(register, output);
-            if (MResetZ(output) == One) {
-                set correct = true;
-                // Read off coloring.
-                set coloring = MeasureColoring(bitsPerColor, register);
-            }
-            ResetAll(register);
-        } until (correct or iter > maxIterations)
-        fixup {
-            set iter += 1;
-        }
-        if (not correct) {
-            fail "Failed to find a coloring.";
-        }
+        Message($"Trying search with {nIterations} iterations...");
+        ApplyGroversAlgorithmLoop(register, oracle, statePrep, nIterations);
+        let res = MultiM(register);
+        let coloring = MeasureColoring(bitsPerColor, register);
```

Comment:
This seems a bit odd, since you just measured in the immediately preceding line; is line 173 still needed, then? It looks like `res` is not used anywhere at the moment.